### ISSUE TITLE
CallbackMask 'None' enum conflict under Linux

### ISSUE THREAD

**Issue by CaptainHorse** [cginternals/glbinding#307]
I'm getting an error when trying to compile a program using glbinding on Ubuntu 18.04 with g++-9

> error: expected identifier before numeric constant
> |     None        = 0,      ///< All callbacks are disabled
> |     ^~~~

After quick search, it would seem that some preprocessor macro could be conflicting with 'None' keyword.

The program compiles fine on MSVC side.

**Comment by scheibel** [comment]
Upon a quick search I found nothing documenting a pre-defined `None` macro in g++9.

However, we experienced the `None` issue in combination with [X11](https://code.woboq.org/qt5/include/X11/X.h.html#_M/None) and GLX headers. Is it possible you have a similar case?
When using glbinding with another windowing toolkit, we suggest to separate compilation units so that those issues are mitigated.
For X11 as well as Qt we use a full separation of compilation units. An example:
 * Qt OpenGL widget: https://github.com/scheibel/graphics-api-test/blob/master/source/graphics-api-test/glwidget.cpp
 * Internally used OpenGL rendering within widget: https://github.com/scheibel/graphics-api-test/blob/master/source/graphics-api-test/rendergl.cpp

When using GLFW, this issue has a slightly simpler workaround, given you won't miss these disabled features:
* Defining `GLFW_INCLUDE_NONE`, e.g. globally for all compilation units as in https://github.com/cginternals/glbinding/blob/master/source/examples/cubescape/CMakeLists.txt#L103

As a more general solution, you can test if including glbinding before any headers defining `None` helps.

**Comment by CaptainHorse** [comment]
I looked around and looks like it is caused by X11 includes.

I'm currently moving my project away from GLFW to have more control over windowing and X11 has caused more name-clashing issues with 'Window' for example. I think I'm going to skip over X11 for now and head straight for Wayland since that doesn't define generic name macros.

Thank you for taking your time to answer and helping me find the solution. Closing issue.
